### PR TITLE
feat: added Document, Flow, DocumentArrayMemmap benchmark

### PR DESCRIPTION
**Pull request by maateen**
# Changes

- moved latency tracking codes to Jina Core
- updated dependency list

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -41,20 +40,50 @@ def _benchmark_import_time() -> Dict[str, float]:
...
-
+        Flow().add().add().add(),
+        Flow().add().add().add(needs='gateway')
+    ]
+    st = time.perf_counter()
     for f in fs:
-        f.post('/', (Document() for _ in range(10000)))
-        f.post('/', (Document(blob=sp.coo_matrix([0, 0, 0, 1, 0])) for _ in range(10000)))
+        with f:
+            f.post(on='/test', inputs=Document())
```
Comment (JoanFM) [inline app.py]: I think we could average by flowing more `requests` per `Flow` and not a single `Document`?

Comment (JoanFM) [reply app.py]: Specially with `dynamic routing` the first request should be slower because sockets are open

**Review by JoanFM — CHANGES_REQUESTED** [review]

File: app.py [hunk 2]
```diff
@@ -41,20 +46,50 @@ def _benchmark_import_time() -> Dict[str, float]:
...
+        Flow().add().add().add(),
+        Flow().add().add().add(needs='gateway')
+    ]
+    st = time.perf_counter()
     for f in fs:
-        f.post('/', (Document() for _ in range(10000)))
-        f.post('/', (Document(blob=sp.coo_matrix([0, 0, 0, 1, 0])) for _ in range(10000)))
+        with f:
+            f.post(on='/', inputs=__doc_generator, request_size=10)
+    flow_time = time.perf_counter() - st
```
Comment (deepankarm) [inline app.py]: This should be measured per flow, right?

Comment (maateen) [reply app.py]: I am confused about the metrics. How can we express the metrics if we measure per flow? What's your suggestion?

**Pull request merged**